pkg/operator/controller/gatewayclass/subscription.go

```diff
@@ -48,18 +53,19 @@ func (r *reconciler) ensureServiceMeshOperatorSubscription(ctx context.Context)
 }
 
 // desiredSubscription returns the desired subscription.
-func desiredSubscription(name types.NamespacedName) (*operatorsv1alpha1.Subscription, error) {
+func (r *reconciler) desiredSubscription(name types.NamespacedName) (*operatorsv1alpha1.Subscription, error) {
 	subscription := operatorsv1alpha1.Subscription{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: name.Namespace,
 			Name:      name.Name,
 		},
 		Spec: &operatorsv1alpha1.SubscriptionSpec{
-			Channel:                "stable",
-			InstallPlanApproval:    operatorsv1alpha1.ApprovalAutomatic,
+			Channel:                r.config.GatewayAPIOperatorChannel,
+			InstallPlanApproval:    operatorsv1alpha1.ApprovalManual,
```

Comment:
What happens to the variable overrides that might be passed in via flags?  Do you need to look for overrides in the env variables at this stage?  Otherwise, why use flag opts?

Reply:
I'm not if I understand the question, but I chose to follow the same pattern as the existing env variables, like `$RELEASE_VERSION` or `$CANARY_IMAGE`. They are also defined as env variables, but actually passed to the operator through flag opts.